aop/src/main/java/io/micronaut/aop/chain/ConstructorInterceptorChain.java

```diff
@@ -181,16 +181,16 @@ public T invoke(T instance, Object... arguments) {
 
         if (beanDefinition instanceof AdvisedBeanType) {
 
-            // intercepted bean constructors include 3 additional arguments in
+            // intercepted bean constructors include 4 additional arguments in
             // addition to the arguments declared in the bean
             // Here we subtract these from the parameters made visible to the interceptor consumer
-            if (originalParameters.length < 3) {
+            if (originalParameters.length < 4) {
                 throw new IllegalStateException("Invalid intercepted bean constructor. This should never happen. Report an issue to the project maintainers.");
             }
             return Arrays.copyOfRange(
                     originalParameters,
                     0,
-                    originalParameters.length - 3
+                    originalParameters.length - 4
```

Comment:
How would this work with old style bean definitions? Couldn't this break with those definitions?